services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/model/ResumableFileDownload.java

```diff
@@ -149,6 +157,15 @@ public OptionalLong totalSizeInBytes() {
         return totalSizeInBytes == null ? OptionalLong.empty() : OptionalLong.of(totalSizeInBytes);
     }
 
+    /**
+     * The lists of parts that were successfully completed and saved to the file. Non-empty only for multipart downloads.
+     *
+     * @return part numbers of a multipart download that were completed saved to file.
+     */
+    public List<Integer> completedParts() {
+        return Collections.unmodifiableList(completedParts);
```

Comment:
Can we wrap with umodifableList in the ctor instead of the getter? Also, we should probably create a new list. Example: https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java#L148

Reply:
will address in full resume PR